src/ss_not_fact.jl

```diff
@@ -33,96 +41,144 @@ function subnormsq(∇ϕx, θ)
+                        -(x[i] - B.μ[i])*s + θ[i]*c
+        end
+    end
+    t + τ, x, θ
+end
+
+
+function sevent(t, x, θ, f, Z::Union{BouncyParticle,Boomerang})
+    t, copy(x), copy(θ), copy(f)
+end
+
 function sticky_pdmp_inner!(Ξ, ∇ϕ!, ∇ϕx, t, x, θ, c, b, t′, f, θf, tfrez, tref, told, (acc, num),
-        Flow::BouncyParticle, κ, args...; strong_upperbounds = false, factor=1.5, adapt=false)
+        Flow::Union{BouncyParticle, Boomerang}, κ, args...; strong_upperbounds = false, factor=1.5, adapt=false)
+    
+    # f[i] is true if i is free
+    # frez[i] is the time to freeze, if free, the time to unfreeze, if frozen
     while true
         tᶠ, i = findmin(tfrez) # could be implemented with a queue
         tt, j = findmin([tref, tᶠ, t′])
         τ = tt - t
-        t, x, θ = move_forward!(τ, t, x, θ, Flow)
+        t, x, θ = smove_forward!(τ, t, x, θ, f, Flow)
         # move forward
         if j == 1 # refreshments of velocities
-            θ, θf = refresh_sticky_vel!(θ, θf, Flow)
+            θ, θf = refresh_sticky_vel!(θ, θf, f, Flow)
             tref = t + waiting_time_ref(Flow) # regenerate refreshment time
             b = ab(x, θ, c, Flow) # regenerate reflection time
             told = t
             t′ = t + poisson_time(b, rand())
-            tfrez = freezing_time!(tfrez, t, x, θ, Flow)
-        elseif j == 2
-            if f[i] # hit 0 -> freeze
+            tfrez = freezing_time!(tfrez, t, x, θ, f, Flow)
+            for i in eachindex(f) # make function later...
+                if !f[i]
+                    tfrez[i] = t - log(rand())/(κ[i]*abs(θf[i]))
+                end
+            end
+        elseif j == 2 # get frozen or unfrozen in i
+            if f[i] # if free
                 if abs(x[i]) > 1e-8
+                    tfrez[i] = freezing_time(x[i], θ[i], Flow) # wrong zero of curve 
                     error("x[i] = $(x[i]) !≈ 0")
                 end
                 x[i] = -0*θ[i]
                 θf[i], θ[i] = θ[i], 0.0 # stop and save speed
                 f[i] = false # change tag
-                # tfrez[i] = t - log(rand())/(κ*abs(θf[i])) #option 1
-                tfrez[i] = t - log(rand())/κ # option 2
+                tfrez[i] = t - log(rand())/(κ[i]*abs(θf[i])) # sticky time
+                # tfrez[i] = t - log(rand()) # option 2
                 if !(strong_upperbounds) #not strong upperbounds, draw new waiting time
                     b = ab(x, θ, c, Flow) # regenerate reflection time
                     told = t
                     t′ = t + poisson_time(b, rand())
                 end
-            else # was frozen ->  unfreeze
+            else # is frozen ->  unfreeze
                 @assert x[i] == 0 && θ[i] == 0
                 θ[i], θf[i] = θf[i], 0.0 # restore speed
                 f[i] = true # change tag
-                tfrez[i] = Inf
+                tfrez[i] = Inf # FIXME Flow isa Boomerang ? π : Inf #freezing_time(x[i], θ[i], Flow)
```

Comment:
```suggestion
                tfrez[i] = t + freezing_time(x[i], θ[i], Flow)
```
@mschauer  saves the day